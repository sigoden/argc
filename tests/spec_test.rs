#[test]
fn test_spec_help() {
    snapshot!(include_str!("spec.sh"), &["spec", "-h"],);
}

#[test]
fn test_spec_cmd_preferred_help() {
    snapshot!(include_str!("spec.sh"), &["spec", "cmd-preferred", "-h"],);
}

#[test]
fn test_spec_cmd_omitted_help() {
    snapshot!(include_str!("spec.sh"), &["spec", "cmd-omitted", "-h"],);
}

#[test]
fn test_spec_cmd_option_names_help() {
    snapshot!(include_str!("spec.sh"), &["spec", "cmd-option-names", "-h"],);
}

#[test]
fn test_spec_cmd_option_formats_help() {
    snapshot!(
        include_str!("spec.sh"),
        &["spec", "cmd-option-formats", "-h"],
    );
}

#[test]
fn test_spec_cmd_option_quotes_help() {
    snapshot!(
        include_str!("spec.sh"),
        &["spec", "cmd-option-quotes", "-h"],
    );
}

#[test]
fn test_spec_cmd_flag_formats_help() {
    snapshot!(include_str!("spec.sh"), &["spec", "cmd-flag-formats", "-h"],);
}

#[test]
fn test_spec_cmd_positional_only_help() {
    snapshot!(
        include_str!("spec.sh"),
        &["spec", "cmd-positional-only", "-h"],
    );
}

#[test]
fn test_spec_cmd_positional_requires_help() {
    snapshot!(
        include_str!("spec.sh"),
        &["spec", "cmd-positional-requires", "-h"],
    );
}

#[test]
fn test_spec_cmd_preferred_exec() {
    snapshot!(
        include_str!("spec.sh"),
        &["spec", "cmd-preferred", "-f", "-o", "A", "AB", "C D"],
    );
}

#[test]
fn test_spec_cmd_option_names_exec() {
    snapshot!(
        include_str!("spec.sh"),
        &[
            "spec",
            "cmd-option-names",
            "--opt2",
            "value2",
            "--opt3",
            "value3_0,value3_1",
            "--opt4",
            "value4_0",
            "--opt4",
            "value4_1",
            "--opt6",
            "a",
        ],
    );
}

#[test]
fn test_spec_cmd_option_names_exec_eval() {
    snapshot!(
        include_str!("spec.sh"),
        &[
            "spec",
            "cmd-option-names",
            "--opt2",
            "value2",
            "--opt3",
            "value3_0,value3_1",
            "--opt4",
            "value4_0",
            "--opt4",
            "value4_1",
            "--opt6",
            "a",
        ],
        eval: true,
    );
}

#[test]
fn test_spec_cmd_positional_with_choices() {
    snapshot!(
        include_str!("spec.sh"),
        &["spec", "cmd-positional-with-choices", "-h"],
    );
}

#[test]
fn test_spec_cmd_positional_with_choices_exec() {
    snapshot!(
        include_str!("spec.sh"),
        &["spec", "cmd-positional-with-choices", "a"],
    );
}

#[test]
fn test_spec_cmd_positional_with_choices_exec_fail() {
    snapshot!(
        include_str!("spec.sh"),
        &["spec", "cmd-positional-with-choices", "x"],
    );
}

#[test]
fn test_spec_cmd_positional_with_choices_and_default() {
    snapshot!(
        include_str!("spec.sh"),
        &["spec", "cmd-positional-with-choices-and-default", "-h"],
    );
}

#[test]
fn test_spec_cmd_positional_with_choices_and_default_exec() {
    snapshot!(
        include_str!("spec.sh"),
        &["spec", "cmd-positional-with-choices-and-default"],
    );
}

#[test]
fn test_spec_cmd_without_any_arg() {
    snapshot!(include_str!("spec.sh"), &["spec", "cmd-without-any-arg"],);
}
