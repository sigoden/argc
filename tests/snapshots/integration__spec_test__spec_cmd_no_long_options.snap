---
source: tests/spec_test.rs
expression: output
---
RUN
spec cmd_no_long_options -h

OUTPUT
cat >&2 <<-'EOF' 
Options without long name

USAGE: spec cmd_no_long_options [OPTIONS] -b <B> -d <D>... -i <I> -l <L>

OPTIONS:
  -a <A>           optional
  -b <B>           required
  -c [<C>...]      optional, multiple
  -d <D>...        required, multiple
  -e <E>           optional, default [default: a]
  -f <F>           choices [possible values: a, b, c]
  -g <G>           choices, default [default: a] [possible values: a, b, c]
  -i <I>           required, choices [possible values: a, b, c]
  -j <J>           optional, default from fn
  -k <K>           choices from fn
  -l <L>           required, choices from fn
  -m <VALUE>       optional, default
  -h, --help       Print help

EOF
exit 0

